src/app/complex/complex-results/complex-navigator/complex-navigator.component.ts

```diff
@@ -15,8 +24,139 @@ export class ComplexNavigatorComponent {
   canRemoveComplexesFromBasket = input<boolean>();
   onComplexRemovedFromBasket = output<string>();
 
-  interactorsSorting = 'Occurrence';
+  componentsSorting = model<NavigatorComponentSorting>();
+  componentsGrouping = model<NavigatorComponentGrouping>();
+  displayType = model<NavigatorDisplayType>();
+  onGroupingChange = output<NavigatorComponentGrouping>();
+  onSortingChange = output<NavigatorComponentSorting>();
+  onDisplayTypeChange = output<NavigatorDisplayType>();
+
   organismIconDisplay = true;
   interactorTypeDisplay = true;
   idDisplay = true;
+
+  _allChanges: OutputRef<any>[] = [
+    this.componentsSorting,
+    this.componentsGrouping,
+    this.displayType
+  ];
+
+  anyChange = output<void>();
+
+  navigatorComponentsWithoutGrouping = computed(() => this.createNavigatorComplexes(this.complexSearch().elements, this.interactors()));
+  navigatorComponentsGroupedByOrthologs = computed(() => this.createOrthologGroups(this.navigatorComponentsWithoutGrouping()));
+  orthologGroupsAvailable = computed(() => this.navigatorComponentsGroupedByOrthologs().some(c => c instanceof NavigatorOrthologGroup));
+  navigatorComponents: INavigatorComponent[] = [];
+
+  constructor(private complexPortalService: ComplexPortalService) {
+    effect(() => {
+      this.setNavigatorComponents(this.navigatorComponentsGroupedByOrthologs(), this.navigatorComponentsWithoutGrouping());
+      this.setIconsDisplay();
+    });
+    this._allChanges.forEach(change => change.subscribe(() => setTimeout(() => this.anyChange.emit())));
+  }
+
+  onGroupingChanged(componentsGrouping: NavigatorComponentGrouping) {
+    this.componentsGrouping.set(componentsGrouping);
```

Comment:
Remove this set call. Same for the set calls in `onSortingChanged` and `onDisplayTypeChanged`. We send the events and then the other components update this, so we don't need it.

Reply:
Ignore, do not remove the set calls. Keep them, but replace the `this.onGroupingChange.emit(componentsGrouping);` call with `this.anyChange.emit();`. Same on the 2 other methods below.